test: pipeline stages suite flakes on Node 20 — workflow ID collision + signal/query race

## Symptom

`pipeline stages` tests intermittently fail on Node 20 with a cascade pattern:

1. First test failure: `AssertionError: expected 'waiting' to equal 'reported'` (or similar stage-status assertion)
2. All subsequent tests in the suite: `WorkflowExecutionAlreadyStartedError: Workflow execution already started`

Node 18, 22, and 24 pass consistently on the same commit. The suite clears on rerun.

Example failures observed:
- `pipeline stages > completes stage when all players report result` — `AssertionError: expected 'waiting' to equal 'reported'`
- `pipeline stages > fails stage immediately on blocker with halt policy` — `AssertionError: expected 'active' to equal 'failed'`

## Occurrence

Hit twice on PR #173 (different first test each time). Cleared on rerun both times. Not a functional regression — pure test infrastructure fragility.

## Root Cause 1 — Workflow ID collision

In `test/helpers.ts`, `startSession()` generates the workflow ID as:

```ts
const workflowId = metadata.isConductor
  ? `claude-session-${metadata.ensemble}-conductor`
  : `claude-session-${metadata.ensemble}-${metadata.playerId}`;
```

All conductor sessions get `claude-session-test-ensemble-conductor` **regardless of `playerId`**. The stages test suite uses `conductorMetadata({ playerId: 'stage-cond-N' })` for each of its 7 tests — but they all resolve to the same workflow ID.

When test N fails before its `destroyUpdate` cleanup runs, the workflow stays running. Test N+1 tries to start the same ID → `WorkflowExecutionAlreadyStartedError` cascade.

**Fix**: Use `playerId` directly for all sessions:
```ts
const workflowId = `claude-session-${metadata.ensemble}-${metadata.playerId}`;
```
Backward-compatible: `conductorMetadata()` defaults `playerId` to `'conductor'`, so callers that don't override get `claude-session-{ensemble}-conductor` unchanged.

## Root Cause 2 — Signal/query timing race

After `await handle.signal(playerReportSignal, {...})`, the signal has been accepted by the Temporal server but the workflow task (processing the signal) may not have run by the time `await handle.query(stagesQuery)` fires. On Node 20's slightly slower event loop, this race surfaces more often.

**Fix**: Add `await skipTime(1)` (already exported from `test/helpers.ts`) after each signal→query pair. Advances Temporal's simulated clock 1ms, flushing pending workflow tasks before the assertion query.

8 callsites in `test/stages.test.ts` need this treatment.

## Proposed fix

Small PR touching only:
- `test/helpers.ts` — remove conductor branch from `workflowId` calculation (3-line change)
- `test/stages.test.ts` — import `skipTime`, add `await skipTime(1)` after each signal that precedes a state assertion (8 callsites)

No product code changes. Fix is already written and validated locally.

## Notes

- Fix was held out of PR #173 to keep test fixes separate from product PRs.
- Both root causes are independent; either alone would reduce flake frequency. Together they should eliminate it.

--- a/test/helpers.ts
+++ b/test/helpers.ts
@@ -244,9 +244,13 @@ export async function startSession(
     ...(inputOverrides.metadata ? {} : { metadata }),
   };
 
-  const workflowId = metadata.isConductor
-    ? `claude-session-${metadata.ensemble}-conductor`
-    : `claude-session-${metadata.ensemble}-${metadata.playerId}`;
+  // Always use playerId in the workflow ID. conductorMetadata() defaults
+  // playerId to 'conductor', so callers that don't override get the canonical
+  // 'claude-session-{ensemble}-conductor' ID unchanged. Callers that pass a
+  // non-default playerId (e.g. stages.test.ts uses 'stage-cond-N') get a
+  // unique ID per test, preventing WorkflowExecutionAlreadyStartedError
+  // cascades when a test fails before its cleanup destroyUpdate runs.
+  const workflowId = `claude-session-${metadata.ensemble}-${metadata.playerId}`;
 
   return testEnv.client.workflow.start('claudeSessionWorkflow', {
     workflowId,

--- a/test/stages.test.ts
+++ b/test/stages.test.ts
@@ -10,7 +10,7 @@ import {
   conductorMetadata,
   withWorker,
   updateMetadataSignal,
-
+  skipTime,
   destroyUpdate,
 } from './helpers';
 import {
@@ -78,6 +78,7 @@ describe('pipeline stages', function () {
         text: 'Tests pass',
         type: 'result',
       });
+      await skipTime(1); // flush pending workflow task before querying
 
       let stages: StageEntry[] = await handle.query(stagesQuery);
       expect(stages[0].status).to.equal('active');
@@ -89,6 +90,7 @@ describe('pipeline stages', function () {
         text: 'All good',
         type: 'result',
       });
+      await skipTime(1); // flush pending workflow task before querying
 
       stages = await handle.query(stagesQuery);
       expect(stages[0].status).to.equal('complete');
@@ -125,6 +127,7 @@ describe('pipeline stages', function () {
         text: 'Build broken',
         type: 'blocker',
       });
+      await skipTime(1); // flush pending workflow task before querying
 
       const stages: StageEntry[] = await handle.query(stagesQuery);
       expect(stages[0].status).to.equal('failed');
@@ -163,6 +166,7 @@ describe('pipeline stages', function () {
         text: 'Found issues',
         type: 'blocker',
       });
+      await skipTime(1); // flush pending workflow task before querying
 
       let stages: StageEntry[] = await handle.query(stagesQuery);
       expect(stages[0].status).to.equal('active'); // Still active with continue policy
@@ -173,6 +177,7 @@ describe('pipeline stages', function () {
         text: 'Looks good',
         type: 'result',
       });
+      await skipTime(1); // flush pending workflow task before querying
 
       stages = await handle.query(stagesQuery);
       expect(stages[0].status).to.equal('failed');
@@ -205,6 +210,7 @@ describe('pipeline stages', function () {
         text: 'What API should I use?',
         type: 'question',
       });
+      await skipTime(1); // flush pending workflow task before querying
 
       const stages: StageEntry[] = await handle.query(stagesQuery);
       expect(stages[0].status).to.equal('active');
@@ -216,6 +222,7 @@ describe('pipeline stages', function () {
         text: 'Done',
         type: 'result',
       });
+      await skipTime(1); // flush pending workflow task before querying
 
       const stages2: StageEntry[] = await handle.query(stagesQuery);
       expect(stages2[0].status).to.equal('complete');
@@ -238,6 +245,7 @@ describe('pipeline stages', function () {
       });
 
       await handle.signal(cancelStageSignal, 'cancelled-stage');
+      await skipTime(1); // flush pending workflow task before querying
 
       const stages: StageEntry[] = await handle.query(stagesQuery);
       expect(stages[0].status).to.equal('cancelled');
@@ -271,6 +279,7 @@ describe('pipeline stages', function () {
         failurePolicy: 'continue',
         createdBy: 'stage-cond-7',
       });
+      await skipTime(1); // flush pending workflow task before querying
 
       const stages: StageEntry[] = await handle.query(stagesQuery);
       expect(stages).to.have.lengthOf(1);

